contracts/contracts/chains/arbitrum/ArbitrumMessengers.sol

```diff
@@ -38,13 +38,13 @@ contract ArbitrumL1Messenger is IMessenger, RestrictedCalls {
         deposits[msg.sender] += msg.value;
     }
 
-    function withdraw() external {
-        uint256 amount = deposits[msg.sender];
+    function withdraw(address target) public {
```

Comment:
passing the address for whom to withdraw is problematic as it is a potential DoS attack vector.
Whenever someone wants or needs to do an L1 resolution, an attacker could frontrun and withdraw his deposit first. The L1 resolution would fail.
A potential prevention could be an atomic execution of deposit and l1 resolution via a proxy (Not saying that this is the best solution, but I think the only one to withdraw on behalf here).

Reply:
Alternatively, we could use tx.origin to authorize like (`require target == target.origin,"not authorized"`) but this is exactly what is known as a potential vulnerability in the contracts, I will outline in a separate comment.